### PR TITLE
Document reconciliation process

### PR DESCRIPTION
**Pull request by ccremer**
## Summary

This should make it somewhat easier for new contributors to understand the mechanics of the provider

## Checklist

- [x] PR contains a single logical change (to keep release notes relevant)
- [x] Categorize the PR by setting a good title and adding one of the labels:
  `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
  as they show up in the changelog
- [x] Link this PR to related issues

### CODE REVIEW

**Review by Kidswiss — COMMENTED** [review]
LGTM

Maybe include the sources for the images? In case we need to adjust things?

**Comment by ccremer** [comment]
> Maybe include the sources for the images? In case we need to adjust things?

They're standard SVG, there should be plenty of SVG editors around. In my case, I used draw.io (diagrams.net), but in VS code (https://marketplace.visualstudio.com/items?itemName=hediet.vscode-drawio)

File: docs/modules/ROOT/pages/explanations/dev/provider-mechanics.adoc [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+NOTE: There is some more boilerplate required to set it up correctly, but not relevant in this page.
+
+The `managed.ExternalConnecter` interface is meant as an entrypoint for every reconciliation.
+It's the step where the runtime fetches some credentials to connect to an external cloud API.
+The return value of `ExternalConnector.Connect()` is an instance of `managed.ExternalClient` itself.
+
+`managed.ExternalClient` features the 4 basic CRUD methods: `Create()`, `Observe()`, `Update()`, `Delete()`.
+By implementing `Connect` and these 4 CRUD methods everything should be in place to get started for a managed CRD.
+
+What is not comprehendably explained is how the reconcilation actually works.
```
Comment (ZuGao) [inline docs/modules/ROOT/pages/explanations/dev/provider-mechanics.adoc:18]: typo `comprehendably` and `reconcilation`

**Review by ZuGao — APPROVED** [review]
LGTM. There are only some typos

**Comment by Kidswiss** [comment]
> > Maybe include the sources for the images? In case we need to adjust things?
> 
> They're standard SVG, there should be plenty of SVG editors around. In my case, I used draw.io (diagrams.net), but in VS code (https://marketplace.visualstudio.com/items?itemName=hediet.vscode-drawio)

Ah I wasn't aware that SVG is the native draw.io format. It's fine then.

**Comment by tobru** [comment]
> Ah I wasn't aware that SVG is the native draw.io format. It's fine then.

I can recommend https://www.vshn.ch/blog/diagrams-with-draw-io/ which made me start using VS Code and Draw.io to create diagrams...

**Pull request merged**